cloud/cloud_env_impl.cc

```diff
@@ -1997,7 +1997,13 @@ size_t CloudEnvImpl::TEST_NumScheduledJobs() const {
 };
 
 Status CloudEnvImpl::ApplyCloudManifestDelta(const CloudManifestDelta& delta) {
-  cloud_manifest_->AddEpoch(delta.file_num, delta.epoch);
+  if (!cloud_manifest_->AddEpoch(delta.file_num, delta.epoch)) {
+    Log(InfoLogLevel::INFO_LEVEL, info_log_,
+        "[CloudEnvImpl::ApplyCloudManifestDelta] delta: %s has already been "
+        "applied to current cloud "
+        "manifest",
+        delta.ToString().c_str());
+  }
```

Comment:
I would prefer we not log inside of RocksDB itself, but rather return the value through `bool* update_applied` and let the caller log if they want to. The caller might also want to fail if they didn't expect `ApplyCloudManifestDelta()` to ignore the update, like we do when upgrading to leader.